src/net.py

```diff
@@ -40,8 +40,9 @@ class Network(object):
 
     def __init__(self, name, dimensions, input_var, y, config,
                  input_network=None, num_classes=None, activation='rectify',
-                 pred_activation='softmax', optimizer='adam',
+                 pred_activation='softmax', optimizer='adam', stopping_rule='best_validation',
```

Comment:
I think we should default it to None so you have to explicitly specify if you want to use early stopping

Reply:
Hmmm....I think 'best on the validation set' should be the default since it will typically have less generalization error than the probably-overfit 'best on the training set'. 

I agree that if/when we add other early stopping options, they should require opting in though, since those could presumably cost you performance.

Reply:
Fair enough, we can keep it that way. not sure how this affects saving and loading functionality. Would you be able to run some tests with that in mind?